code/game/objects/items/devices/instruments.dm

```diff
@@ -268,3 +268,45 @@
 	item_state = "electronic"
 	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+
+/obj/item/musicbeacon
+	name = "express delivery beacon"
+	desc = "Summon your tool of art."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-red"
+	item_state = "radio"
+
+/obj/item/musicbeacon/attack_self(mob/user)
+	beacon_music(user)
+
+/obj/item/musicbeacon/proc/beacon_music(mob/M)
+	var/list/instruments = list(
+								/obj/item/instrument/violin,
+								/obj/item/instrument/piano_synth,
+								/obj/item/instrument/guitar,
+								/obj/item/instrument/eguitar,
+								/obj/item/instrument/glockenspiel,
+								/obj/item/instrument/accordion,
+								/obj/item/instrument/trumpet,
+								/obj/item/instrument/saxophone,
+								/obj/item/instrument/trombone,
+								/obj/item/instrument/recorder,
+								/obj/item/instrument/harmonica)
+	var/list/display_names = list()
+	for(var/V in instruments)
+		var/atom/A = V
+		display_names += list(initial(A.name) = A)
+
+	var/choice = input(M,"What instrument would you like to order?","Jazz Express") as null|anything in display_names
+	if(QDELETED(src) || !choice || M.stat || !in_range(M, src) || M.restrained() || !M.canmove)
+		return
+
+	var/index = display_names.Find(choice)
+	var/A = instruments[index]

```

Comment:
Depending on if calling a 0 index is an issue in BYOND, you may want to simply check if the index is 0 and return here instead of checking for instrument on line 310.

From what I can ascertain from BYOND docs, the Find proc will return 0 if it can't find the given element in the list.

Reply:
>starting lists at one
Byond Y